src/components/LayoutTree.tsx

```diff
@@ -909,7 +909,11 @@ const LayoutTree = () => {
   ])
 
   const spaceAboveLast = useRef(spaceAboveExtended)
-
+  // When the cursor is in a table, all thoughts beneath the table are hidden,
```

Comment:
It might also be worth mentioning in this comment the reason that this is needed. i.e. Why is col2 a special case for the `hideCaret` depth?